src/edu/washington/escience/myria/CsvTupleWriter.java

```diff
@@ -57,9 +62,20 @@ public void writeColumnHeaders(final List<String> columnNames) throws IOExceptio
   public void writeTuples(final ReadableTable tuples) throws IOException {
     final String[] row = new String[tuples.numColumns()];
     /* Serialize every row into the output stream. */
+    Schema tbsc = tuples.getSchema();
     for (int i = 0; i < tuples.numTuples(); ++i) {
       for (int j = 0; j < tuples.numColumns(); ++j) {
-        row[j] = tuples.getObject(j, i).toString();
+        Type type = tbsc.getColumnType(j);
+        if (type.equals(Type.BLOB_TYPE)) {
+          // write the file out
+          // add filename to the csv file
+          String filename = UUID.randomUUID().toString();
```

Comment:
Strongly disagree; IMO we should rely on the core language API by default, and only roll-our-own implementation if there is a strong reason to do otherwise (this is uncommon).  The code would also be strengthened by using Java-flavored RAII and automatically cleaning up the temp files (`deleteOnExit`).